test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,130 @@
+"use strict";
+
+require('./mockmobile');
+var fs = require('fs-extra-promise');
+var path = require('path');
+var crypto = require('crypto');
+var tape = require('wrapping-tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+// get the values needed for running the tests
+var configValues = cryptomanager.getConfigValuesForTestingOnly();
+
+var fileLocation = './pkcs12folder';
+
+// test setup & teardown activities
+var test = tape({
+  setup: function(t) {
+    fs.mkdirSync(fileLocation);
+    t.end();
+  },
+  teardown: function(t) {
+    fs.remove(fileLocation, function(err) {
```

Comment:
So... um.... just out of curiosity... (don't you dare fix this!) But why would you use fs.mkdirSync on line 18 and then use fs.remove instead of fs.removeSync on line 22? Again, DON'T FIX THIS! But doesn't this just feel wrong to you? Don't you want your code to be internally consistent? You now make the developer wonder - why did he use a sync fs function and then 4 lines later use an async one? In this case blocking on sync is find since the tests are going to block anyway (they are linear) and you could completely skip the if(err) step since the sync version will throw an exception. 

to be clear, your code is correct. It will work. This isn't worth fixing. My only point in sending this message is to try and show examples where your developer instincts should 'ping' and bug you.